test/integration/test_gatewayconfiguration_service_name.go

```diff
@@ -0,0 +1,122 @@
+				Network: operatorv1beta1.GatewayConfigDataPlaneNetworkOptions{
+					Services: &operatorv1beta1.GatewayConfigDataPlaneServices{
+						Ingress: &operatorv1beta1.GatewayConfigServiceOptions{
+							ServiceOptions: operatorv1beta1.ServiceOptions{
+								Name: &customServiceName,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	gatewayConfig, err := GetClients().OperatorClient.GatewayOperatorV1beta1().GatewayConfigurations(namespace.Name).Create(GetCtx(), gatewayConfig, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cleaner.Add(gatewayConfig)
+
+	t.Log("deploying a GatewayClass resource with the GatewayConfiguration attached via ParametersReference")
+	gatewayClass := &gatewayv1.GatewayClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: uuid.NewString(),
+		},
+		Spec: gatewayv1.GatewayClassSpec{
+			ParametersRef: &gatewayv1.ParametersReference{
+				Group:     gatewayv1.Group(operatorv1beta1.SchemeGroupVersion.Group),
+				Kind:      gatewayv1.Kind("GatewayConfiguration"),
+				Namespace: (*gatewayv1.Namespace)(&gatewayConfig.Namespace),
+				Name:      gatewayConfig.Name,
+			},
+			ControllerName: gatewayv1.GatewayController(vars.ControllerName()),
+		},
+	}
+	gatewayClass, err = GetClients().GatewayClient.GatewayV1().GatewayClasses().Create(GetCtx(), gatewayClass, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cleaner.Add(gatewayClass)
+
+	t.Log("deploying Gateway resource")
+	gateway := &gwtypes.Gateway{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace.Name,
+			Name:      uuid.NewString(),
+		},
+		Spec: gatewayv1.GatewaySpec{
+			GatewayClassName: gatewayv1.ObjectName(gatewayClass.Name),
+			Listeners: []gatewayv1.Listener{{
+				Name:     "http",
+				Protocol: gatewayv1.HTTPProtocolType,
+				Port:     gatewayv1.PortNumber(80),
+			}},
+		},
+	}
+	gateway, err = GetClients().GatewayClient.GatewayV1().Gateways(namespace.Name).Create(GetCtx(), gateway, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cleaner.Add(gateway)
+
+	t.Log("verifying that the DataPlane has the custom service name")
+	require.Eventually(t, func() bool {
+		dataplanes, err := gatewayutils.ListDataPlanesForGateway(GetCtx(), GetClients().MgrClient, gateway)
+		if err != nil {
+			return false
+		}
+		if len(dataplanes) != 1 {
+			return false
+		}
+		dp := dataplanes[0]
+		if dp.Spec.Network.Services == nil || dp.Spec.Network.Services.Ingress == nil || dp.Spec.Network.Services.Ingress.Name == nil {
+			return false
+		}
+		return *dp.Spec.Network.Services.Ingress.Name == customServiceName
+	}, testutils.GatewayReadyTimeLimit, time.Second)
+
+	t.Log("verifying that the service has the custom name")
+	require.Eventually(t, func() bool {
+		var serviceList corev1.ServiceList
+		err := GetClients().MgrClient.List(GetCtx(), &serviceList, client.InNamespace(namespace.Name))
+		if err != nil {
+			return false
+		}
+		for _, svc := range serviceList.Items {
+			if svc.Name == customServiceName {
+				return true
+			}
+		}
```

Comment:
Instead of listing all `Service`s we can use `DataPlane`'s `status.service` field https://github.com/Kong/kubernetes-configuration/blob/27a5add0384e9758bca11e8a849409ab86ef6e48/api/gateway-operator/v1beta1/dataplane_types.go#L318 which indicates which `Service` the `DataPlane` exposes.